cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -337,13 +339,13 @@ public void runPipeline(PhaseSpec phaseSpec, String sourcePluginType,
                                                     stageSpec, functionCacheFactory, plugin,
                                                     numPartitions, collector, shufflers);
         addEmitted(emittedBuilder, pipelinePhase, stageSpec,
-                   joined.map(new RecordInfoWrapper<>(stageName)), groupedDag, branchers, shufflers, false, false);
+                   mapToRecordInfoCollection(stageName, joined), groupedDag, branchers, shufflers, false, false);
 
       } else if (Windower.PLUGIN_TYPE.equals(pluginType)) {
 
         Windower windower = pluginContext.newPluginInstance(stageName, macroEvaluator);
         SparkCollection<Object> windowed = stageData.window(stageSpec, windower);
-        addEmitted(emittedBuilder, pipelinePhase, stageSpec, windowed.map(new RecordInfoWrapper<>(stageName)),
+        addEmitted(emittedBuilder, pipelinePhase, stageSpec, mapToRecordInfoCollection(stageName, windowed),
```

Comment:
seems like this shouldn't be needed in the windower section?

Reply:
It's not really needed, however since we already do this for the Joiner plugin, it seems that it would be wise to use the same function (as it allows us to add additional future functionality, or make changes in only one place).